### DISCUSSION TITLE
Why rustdesk isn't working offline anymore?

### DISCUSSION THREAD

**Discussion by Hatibovics** [rustdesk/rustdesk#6118] [Q&A]
I've installed rustdesk more than a year ago (version 1.1.9) and now it's not working offline, by this I mean that I've two laptops connected to the same router but the router isn't connected to a modem, back then when I installed it it worked, I just installed the software, set passwords for the two laptops and that's it, I didn't do anything else in the program and it worked offline, but now it requires internet connection, why?

Edit: I connected to the other laptop while being online, rustdesk said that it's a direkt connection, so it should work offline, so I disconnected the router from the modem and it's working offline, but then I disconnected from the other laptop and tried to reconnect offline and this time it can't connect, it seems that it only have to be online to start the connection, but why it can't start the connection offline?

**Comment by My1** [comment]
maybe it only works if you enter your IP as it doesnt seem to scan for nearby devices

**Reply by Hatibovics** [reply]
I know that there is an option for that, but I read somewhere that direct IP connection is not safe, I use these laptops offline and online, so it would be great if Rustdesk would work both ways without changing between direct and non-direct IP access.

**Reply by My1** [reply]
well direct IP can be a not so great idea in the wider internet but if you are offline you can just try entering the IP in the local network (usually 192.168.x.y, there are a few other options tho) to connect.

generally firewalls of most routers (and on IPv4 additionally NAT) block incoming random connections
